identity-actor/src/actor/errors.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use libp2p::request_response::OutboundFailure;
+
+use crate::didcomm::ThreadId;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, thiserror::Error)]
+/// Errors that can occur during the actor operation.
+pub enum Error {
+  #[error("{context}: {source}")]
+  TransportError {
+    context: &'static str,
+    source: libp2p::TransportError<std::io::Error>,
```

Comment:
Unfortunately this exposes a type from an unstable external crate in our public API. Also the error message should not contain the error message of source (when a field is named source `thiserror` will automatically return it via `Error::source`). 
 
```suggestion 
#[non_exhaustive]
#[error("transport error: {0}")]
TransportError(&'static str, #[source] libp2p::TransportError<std::io::Error>), 
```